fdbrpc/TokenCache.cpp

```diff
@@ -0,0 +1,98 @@
+#include "fdbrpc/FlowTransport.h"
+#include "fdbrpc/TokenCache.h"
+#include "fdbrpc/TokenSign.h"
+#include "flow/network.h"
+
+#include <boost/compute/detail/lru_cache.hpp>
+#include <boost/unordered_set.hpp>
+
+struct TokenCacheImpl {
+	struct CacheEntry {
+		Arena arena;
+		boost::unordered_set<TenantNameRef> tenants;
+		double expirationTime = 0.0;
+	};
+
+	boost::compute::detail::lru_cache<StringRef, CacheEntry> cache;
+	TokenCacheImpl() : cache(FLOW_KNOBS->TOKEN_CACHE_SIZE) {}
+
+	bool validate(TenantNameRef tenant, StringRef token);
+};
+
+TokenCache::TokenCache() : impl(new TokenCacheImpl()) {}
+TokenCache::~TokenCache() {
+	delete impl;
+}
+
+void TokenCache::createInstance() {
+	g_network->setGlobal(INetwork::enTokenCache, new TokenCache());
+}
+
+TokenCache& TokenCache::instance() {
+	return *reinterpret_cast<TokenCache*>(g_network->global(INetwork::enTokenCache));
+}
+
+bool TokenCache::validate(TenantNameRef name, StringRef token) {
+	return impl->validate(name, token);
+}
+
+bool TokenCacheImpl::validate(TenantNameRef name, StringRef token) {
+	auto sig = authz::jwt::signaturePart(token);
```

Comment:
Parse token is much more expensive, since we're now doing this for each request, I think parsing the json and doing the base64 conversion (plus all the allocation that come with it) are too expensive. In the `else`-case this should still fail.